src/state_machine/signer/mod.rs

```diff
@@ -180,7 +180,7 @@ impl<SignerType: SignerTrait> Signer<SignerType> {
         public_keys: PublicKeys,
     ) -> Self {
         assert!(threshold <= total_keys);
-        let mut rng = OsRng;
+        let mut rng = create_rng();
```

Comment:
Instead of calling `create_rng` here, please pass the `rng` in as a parameter to `new`